modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/ReciprocalRank.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.rankeval;
+
+import org.elasticsearch.common.ParseFieldMatcherSupplier;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.rankeval.PrecisionAtN.Rating;
+import org.elasticsearch.index.rankeval.PrecisionAtN.RatingMapping;
+import org.elasticsearch.search.SearchHit;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.naming.directory.SearchResult;
+
+/**
+ * Evaluate reciprocal rank.
+ * */
+public class ReciprocalRank extends RankedListQualityMetric {
+
+    public static final String NAME = "reciprocal_rank";
+    // the rank to use if the result list does not contain any relevant document
+    // TODO decide on better default or make configurable
+    private static final int RANK_IF_NOT_FOUND = Integer.MAX_VALUE;
```

Comment:
I believe https://www.wikipendium.no/TDT4117_Information_retrieval has a better explanation of what to do if the document is not found at all.

The link above also states that the user should specify which is the largest acceptable rank. I think it makes sense to explicitly let users configure this: Depending on how many results are shown on the result page this number may vary. E.g. think of the difference of a mobile vs. a desktop result page: In the former case users are less likely to click results they have to scroll down to, while in the latter case those results might be readily visible without any user interaction.